### PR TITLE
fix userprog

### PR DESCRIPTION
**Pull request by ZeroWu1412**

### CODE REVIEW

**Review by bmwfire — APPROVED** [review]
This looks good to me! It's interesting how you created a new function to manually close files if the list is not empty. I'll let either Charles or Andrew review this before I merge. Well done, Zero!

File: pintos/src/userprog/process.h [hunk 1]
```diff
@@ -2,8 +2,33 @@
...
+/* States of a file that has been loaded to be run by a process. */
+enum load_status
+  {
+    NOT_LOADED,         /* Initial loading state. */
+    LOAD_SUCCESS,       /* The file was loaded successfully with no issues. */
+    LOAD_FAILED         /* The file failed to load. */
+  };
+
+/* A structure storing infomation about a child process. */
+struct process
```
Comment (dickensc) [inline pintos/src/userprog/process.h:19]: defined but never used?

**Review by dickensc — APPROVED** [review]

File: pintos/src/userprog/process.h (hunk 1 above)
Comment (ZeroWu1412) [reply pintos/src/userprog/process.h:19]: I was trying all kind of stuff and that was part of it (I have like 16 branch in total). Free feel to delete, it was affect the code

Comment (ZeroWu1412) [reply pintos/src/userprog/process.h:19]: wont affected the code I mean.

**Pull request merged**